src/OpcUaStackServer/ServiceSet/MonitorManager.cpp

```diff
@@ -582,8 +582,9 @@ namespace OpcUaStackServer
 		}
 
 		// check monitored item list
+		monitoredItemIds_.erase(it1);
 		if (newMonitoredItemIds.size() > 0) {
-			monitoredItemIds_.insert(std::make_pair(baseNodeClass->nodeId().data(), monitoredItemIds));
+			monitoredItemIds_.insert(std::make_pair(baseNodeClass->nodeId().data(), newMonitoredItemIds));
 			return;
```

Comment:
Oh , the same data will be inserted again.

Reply:
Not the same. `newMonitoredItemIds` has the rest of ID's and I re-insert them deleting old ID's before. See L585.